macros/src/path.rs

```diff
@@ -0,0 +1,244 @@
+
+impl CustomPath {
+    
+    pub fn get_path_and_some_decl(&self, ts_name: &String ) -> FnOutputPathBody {
+
+        match self {
+
+            Self::Str(input) => { Self::str_path(input,ts_name) },
+
+            Self::Static(input) => { Self::static_path(input,ts_name) },
+
+            Self::Fn(input) => { Self::fn_path(input,ts_name) },
+
+            Self::Env(input) => { Self::env_path(input,ts_name) },
+
+        } 
+    }
+
+    fn str_path( input: &String, ts_name: &String ) -> FnOutputPathBody {
+
+        let path = 
+        if input.ends_with('/') {
+            format!("{}{}.ts", input, ts_name)
+        }  else {
+            input.to_owned()
+        };
+
+        return (quote!(#path),None);
+    }
+
+    fn static_path( input: &syn::Path, ts_name: &String ) -> FnOutputPathBody {
+
+        let path_ident      = format_ident!("path");
+        let stat_path_ident = format_ident!("PATH");
+        let path_decl = quote! {
+
+            static #stat_path_ident: std::sync::OnceLock<String> = std::sync::OnceLock::new();
+        
+            let #path_ident = #stat_path_ident.get_or_init( ||
+                {
+                    if #input.ends_with('/')  {
+                        format!("{}{}.ts", #input, #ts_name)
+                    } else {
+                        format!("{}",#input)
+                    }
+                } 
+            );
+        };
+
+        ( quote!(#path_ident), Some(path_decl) )
+    }
+
+    fn fn_path( input: &syn::Path, ts_name: &String ) -> FnOutputPathBody { 
+        
+        ( quote!{#input (#ts_name)?}, None)
+    }
+
+    fn env_path( input: &LitStr, ts_name: &String ) -> FnOutputPathBody {
+
+        let path_ident = format_ident!("path");
+
+        let path_decl = quote!{
+
+            let #path_ident = if std::env!(#input).ends_with('/') {
+                std::concat!(std::env!(#input),#ts_name,".ts")
+            } else {
+                std::env!(#input)
+            };
+        };
+
+        ( quote!(#path_ident), Some(path_decl) )
+    }
+
+}
+
+impl Parse for CustomPath {
+
+    fn parse(input: ParseStream) -> Result<CustomPath> {
+        input.parse::<Token![=]>()?;
+        let span = input.span();
+
+        let msg = 
+"expected arguments for 'export_to':
+
+1) string literal 
+    #[ts(export_to = \"my/path\")] 
+
+2) static or constant variable name 
+
+    #[ts(export_to = MY_STATIC_PATH)]
+    #[ts(export_to = crate::MY_STATIC_PATH)]
+
+Note: This option is available for Rust 1.7.0 and higher!
+
+3) function name of a `Fn(&'static str) -> Option<&'static str>`
+
+    #[ts(export_to = get_path)]
+    #[ts(export_to = crate::get_path)]
+
+Note: This option overrides the original `TS::output_path` logic`!
+
+4) environment variable name  
+
+    #[ts(export_to = env(\"MY_ENV_VAR_PATH\"))] 
+
+Note: This option is for environment variables defined in the '.cargo/config.toml' file only, accessible through the `env!` macro!
+";
```

Comment:
This message should be a module level `const` to avoid having a massive string literal inside the function. This makes the function more difficult to read